csi/controller.go

```diff
@@ -587,14 +586,17 @@ func (s *OsdCsiServer) CreateVolume(
 		}
 		newVolumeId = createResp.VolumeId
 	} else {
-		clonedMetadata := getClonedPVCMetadata(locator)
+		labels := locator.GetVolumeLabels()
 		if spec.GetFADAPodName() != "" {
-			clonedMetadata[sdk.FADAPodLabelKey] = spec.GetFADAPodName()
+			labels[api.SpecPurePodName] = spec.GetFADAPodName()
 		}
+		delete(labels, intreePvcNameKey)
+		delete(labels, intreePvcNamespaceKey)
+		delete(labels, api.SpecParent)
```

Comment:
Can you help me understand why do we need to remove this label here?

Reply:
The PVC name and namespace keys don't make sense for a clone, as they are no longer attached to the same PVC and namespace.

Deleting the parent label was to make the tests happy (if it wasn't present before I wasn't sure if we wanted it present now), but I can change that to still be present. I don't _think_ it breaks any behavior to have it there.

Reply:
Yes my concern was with the parent label only. If it's not being used further then I guess it's ok to remove it.

Reply:
Actually was able to remove this chunk!